src/main/java/past/cwdt/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -1,29 +1,39 @@
 package past.cwdt.bookstore.service.impl;
 
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
 import lombok.RequiredArgsConstructor;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 import past.cwdt.bookstore.dto.book.BookDto;
 import past.cwdt.bookstore.dto.book.BookDtoSearchByParameters;
+import past.cwdt.bookstore.dto.book.BookDtoWithoutCategoryIds;
 import past.cwdt.bookstore.dto.book.CreateBookRequestDto;
 import past.cwdt.bookstore.exceptions.EntityNotFoundException;
 import past.cwdt.bookstore.mapper.BookMapper;
 import past.cwdt.bookstore.model.Book;
+import past.cwdt.bookstore.model.Category;
 import past.cwdt.bookstore.repository.BookRepository;
+import past.cwdt.bookstore.repository.book.CategoryRepository;
 import past.cwdt.bookstore.repository.impl.BookSpecificationBuilder;
 import past.cwdt.bookstore.service.BookService;
 
 @RequiredArgsConstructor
 @Service
+@Transactional
```

Comment:
You better use it with a method that requires this annotation and you can add readOnly property to get()